src/server/src/main/java/io/cassandrareaper/resources/RepairRunResource.java

```diff
@@ -379,22 +392,33 @@ public Response modifyRunState(
           .entity("invalid \"state\" argument: " + state.get())
           .build();
     }
-    final RepairRun.RunState oldState = repairRun.get().getRunState();
+    final RepairRun.RunState oldState = repairRun.getRunState();
+
+    if (intesityStr.isPresent() && oldState == newState && newState == RepairRun.RunState.PAUSED) {
```

Comment:
is `intesityStr.isPresent()` required in the conditional…?

Reply:
Yes, because you can use REST API. For example, make two PUT calls with PAUSED state and with no intensity. For this case response should be 301 Not Modified, right?
This condition means that "you can set intensity but it will be used in case when repair run was paused and you don't change state in this during this request".